fs-storage/src/folder_storage.rs

```diff
@@ -0,0 +1,573 @@
+use serde::{Deserialize, Serialize};
+use std::fs::{self, File};
+use std::io::{Read, Write};
+use std::time::SystemTime;
+use std::{
+    collections::BTreeMap,
+    path::{Path, PathBuf},
+};
+
+use crate::base_storage::{BaseStorage, SyncStatus};
+use crate::monoid::Monoid;
+// use crate::utils::read_version_2_fs;
+use data_error::{ArklibError, Result};
+
+/*
+Note on `FolderStorage` Versioning:
+
+`FolderStorage` is a basic key-value storage system that persists data to disk.
+where the key is the path of the file inside the directory.
+
+
+In version 2, `FolderStorage` stored data in a plaintext format.
+Starting from version 3, data is stored in JSON format.
+
+For backward compatibility, we provide a helper function `read_version_2_fs` to read version 2 format.
+*/
```

Comment:
This versioning information is specific to `FileStorage`. It is unrelated to `FolderStorage`.

Since this is our first iteration of `FolderStorage`, we should begin with version 1.

Reply:
We can remove all versioning related information since we do need it right now since our one and default serde format is bincode (which is optimized for fast serde).

We can later add versioning information as a drop in by inserting a `.version` file into the folder with specific custom serde format. If it doesn't exist, the implementation can default to bincode.

Reply:
Yeah, we don't need versioning of value files here, because serialization is provided by the user.